shared/sszutil/deep_equal.go

```diff
@@ -28,14 +29,13 @@ type visit struct {
 // Tests for deep equality using reflected types. The map argument tracks
 // comparisons that have already been seen, which allows short circuiting on
 // recursive types.
-func deepValueEqual(v1, v2 reflect.Value, visited map[visit]bool, depth int) bool {
+func deepValueEqual(v1, v2 reflect.Value, visited map[visit]bool, depth int, unexportedOnly bool) bool {
```

Comment:
Instead of passing in a bool flag, have you thought of the two functions approach?

`deepValueEqual` and `deepValueEqualUnexportedOnly`

Reply:
Tried it! Review and let me know what you think!